runtime/runtime-core/common/src/aws/smithy/kotlin/runtime/collections/CaseInsensitiveString.kt

```diff
@@ -0,0 +1,11 @@
+package aws.smithy.kotlin.runtime.collections
+
+internal class CaseInsensitiveString(val original: String) {
```

Comment:
Missing tests for CaseInsensitiveString. It is tested indirectly through the other classes though

Reply:
Tested indirectly but you're right, it'd be better to have dedicated tests. Will add!